### ISSUE TITLE
titan push return 'Ljava/io/FileNotFoundException;'

### ISSUE THREAD

**Issue by LetItRain** [titan-data/titan#119]
i used command ` titan push -r origin mongo-test`  to push data to Ubuntu through ssh(rsync). then return `Ljava/io/FileNotFoundException;`
please tell me how to fix it.

**Comment by eschrock** [comment]
Can you try with TITAN_DEBUG, via the following?

```
$ TITAN_DEBUG=true titan push -r origin mongo-test
```

Also, is it possible to attach the results of `docker logs titan-server`?

**Comment by LetItRain** [comment]
@eschrock 
Hi eschrock，
 Thank you for your reply, I have solved the previous problem, but there is a new problem, and it fails when push remote.
```
Command failed: Warning: Permanently added 'host' (ECDSA) to the list of known hosts.
bash: /home/ubuntu/mongo-test/3b009538f626438d8ee90a6d2c8c1d88/metadata.json: Permission denied
```

Push failed`

**Comment by eschrock** [comment]
Thanks, glad you sorted it out! Would love to know what the issue was, since that error message could certainly be improved.

 For the new error, I think I know what the issue might be, in that we're sloppy in mixing the use of sudo. For rsync, we have to run as root so that we can chown files properly. And we run `mkdir -p` under sudo as well. But it looks like we don't use sudo when creating or reading the metadata file. We need to pick a side and stick to it: either we should only use sudo within the volume directories, or we should use it everywhere, metadata included.

I *think* the fix would be to change our call to `sudo mkdir` to just `mkdir`, and leave `sudo` only to the rsync effort. Let me see if I can recreate the issue with an unprivileged account (our end2end tests currently use root).

**Comment by LetItRain** [comment]
`FileNotFoundException` Because the path of the ssh key is wrong. shy :)

**Comment by LetItRain** [comment]
@eschrock 
Hi eschrock , i used root user to push, and its successful. But it seems less secure to use the root user in a commercial environment. Is there any other way to solve it？or will this problem be resolved in the next release?
Best Regards

**Comment by eschrock** [comment]
Yes, this is definitely not secure, and we'll address it in the next release. Our primary focus at the moment has been to get the core workflows vetted in the community, but there's a lot of work to be done to make more robust remote providers. For example, the SSH provider as-is won't ever scale because it release on reading "*/*.metadata.json" for many operations. We'll ultimately build a remote server that can have a real database for metadata, access controls, etc. But for now we're keeping it simple with SSH and S3. But we'll still fix this as it's an easy change and a big step up for security!

**Comment by LetItRain** [comment]
ssh clone failed
`titan clone -p keyFile=C:\ssh\[key] ssh://root@[host]/home/ubuntu/mongo-test mongo-test`

```
Creating repository mongo-test
Creating docker volume mongo-test/v0 with path /data/configdb
Creating docker volume mongo-test/v1 with path /data/db
Command failed: docker run --label io.titandata.titan --mount type=volume,src=mongo-test/v0,dst=/data/configdb,volume-driver=titan --mount type=volume,src=mongo-test/v1,dst=/data/db,volume-driver=titan --name mongo-test --label io.titandata.titan --mount type=volume,src=mongo-test/v1,dst=/data/db,volume-driver=titan -p 27017:27017 -d mongo@sha256:1a9478d8188d6be31dd2e8de076d402edf20446e54933aad7ff49f5b457d486c
docker: Error response from daemon: Duplicate mount point: /data/db.
See 'docker run --help'.

Deleting volume mongo-test/v0
Deleting volume mongo-test/v1
mongo-test removed
```

**Comment by mcred** [comment]
@LetItRain Which version of titan and OS are you using? This does look like a bug that was patched with titan-data/titan#98

**Comment by LetItRain** [comment]
@mcred 
OS: windows server 2019 
docker: 19.03.5
titan: 0.3.5
